README


This archive contains the distribution Dist-Zilla-Plugin-Test-Kwalitee,
version 2.11:

  Release tests for kwalitee

This software is copyright (c) 2011 by Karen Etheridge.

This is free software; you can redistribute it and/or modify it under
the same terms as the Perl 5 programming language system itself.


This README file was generated by Dist::Zilla::Plugin::Readme v5.020.



Changes
Revision history for Perl extension Dist-Zilla-Plugin-Test-Kwalitee

2.11      2014-08-27 01:40:14Z
    - fix test failures with new metrics added in MCA 0.93_01

2.10      2014-08-23 17:55:26Z
    - fix pod error (thanks haarg!)
    - generated test filename is now configurable

2.09      2014-08-23 08:40:01Z
    - sort the arguments to kwalitee_ok, for smaller diffs between iterations
    - now again running the generated test during tests, but without
      attempting to capture its output cleverly

2.08      2014-08-16 04:56:06Z
    - line numbers in shipped code are now almost the same (within 3) as
      the repository source, for easier debugging
    - use kwalitee_ok() interface newly added in Test::Kwalitee 1.21

2.07      2013-10-15 02:18:02Z
    - include in the generated test the exact version of the plugin that
      generated the test
    - warnings tests bypassed during installation
    - update minimum required version of Test::Kwalitee

2.06      2013-08-16 19:41:24 UTC-0700
    - removed old and deprecated Dist::Zilla::Plugin::KwaliteeTests, which is
      no longer indexed by PAUSE anyway
    - fixed tests broken on win32
    - Test::Kwalitee is now required to be present when the test is run (by
      authors, at release time)
    - cleaned up prerequisite list

2.05      2013-08-11 14:18:58 UTC-0700
    - adjust tests to handle recent changes to Test::Kwalitee and
      Module::CPANTS::Analyse

2.04      2013-07-25 17:29:10 UTC-0700
    - fix installation failures with Test::Kwalitee 1.09 (issue #4,
      harleypig's repo)

2.03      2011-10-19 08:56:47 UTC-0600

2.020000  2011-09-07 20:47:05 UTC-0600
    - Provide a deprecation warning (Mike Doherty)
    - Moved switch_namespace branch to master ( HarleyPig )
    - brought dist.ini up to date ( HarleyPig )

1.112410  2011-08-29 09:57:31 UTC-0600
    - Made pod coverage test pass ( HarleyPig )
    - Fixed error with tests needing 'plan' defined ( Kent Fredric)
    - Replace 2-phase inject/munge with 1-phase inject ( Kent Fredric )
    - Add some tests for our test being run ( Kent Fredric )

1.112400  2011-08-28 09:34:37 UTC-0600

1.112390  2011-08-26 21:11:01 UTC-0600

1.101420  2010-05-22 17:47:26 UTC+0200
    - removed weaver.ini since that's handled in Dist::Zilla's [@MARCEL] now
    - list Test::Kwalitee in dist.ini so we don't use() it anymore

1.100690  2010-03-10 23:43:35 UTC+0100
    - original version


Build.PL
# This Build.PL for Dist-Zilla-Plugin-Test-Kwalitee was generated by
# Dist::Zilla::Plugin::ModuleBuildTiny::Fallback 0.005
if (eval 'use Module::Build::Tiny 0.037; 1')
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";

    # This section for Dist-Zilla-Plugin-Test-Kwalitee was generated by Dist::Zilla::Plugin::ModuleBuildTiny 0.006.
    use 5.006;
    # use Module::Build::Tiny 0.037;
    Build_PL();
}
else
{
    $ENV{PERL_MB_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are running this file manually, in which case you need
to learn to first fulfill all configure requires prerequisites listed in
META.yml or META.json -- or use a cpan client to install this distribution.

You can also silence this warning for future installations by setting the
PERL_MB_FALLBACK_SILENCE_WARNING environment variable, but please don't do
that until you fix your toolchain as described above.

EOW
    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));


    # This section was automatically generated by Dist::Zilla::Plugin::ModuleBuild v5.020.
    use strict;
    use warnings;

    require Module::Build; Module::Build->VERSION(0.28);


    my %module_build_args = (
      "build_requires" => {},
      "configure_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "Module::Build::Tiny" => "0.037"
      },
      "dist_abstract" => "Release tests for kwalitee",
      "dist_author" => [
        "Harley Pig <harleypig\@gmail.com>",
        "Caleb Cushing <xenoterracide\@gmail.com>",
        "Marcel Gruenauer <marcel\@cpan.org>",
        "Kent Fredric <kentfredric\@gmail.com>",
        "Karen Etheridge <ether\@cpan.org>"
      ],
      "dist_name" => "Dist-Zilla-Plugin-Test-Kwalitee",
      "dist_version" => "2.11",
      "license" => "perl",
      "module_name" => "Dist::Zilla::Plugin::Test::Kwalitee",
      "recommends" => {},
      "recursive_test_files" => 1,
      "requires" => {
        "Data::Section" => "0.004",
        "Dist::Zilla::File::InMemory" => 0,
        "Dist::Zilla::Plugin::MetaResources" => "4.300039",
        "Dist::Zilla::Role::FileGatherer" => 0,
        "Dist::Zilla::Role::PrereqSource" => 0,
        "Dist::Zilla::Role::TextTemplate" => 0,
        "Moose" => 0,
        "Sub::Exporter::ForMethods" => 0,
        "namespace::autoclean" => 0,
        "perl" => "5.006",
        "strict" => 0,
        "warnings" => 0
      },
      "script_files" => [],
      "test_requires" => {
        "ExtUtils::MakeMaker" => 0,
        "File::Spec" => 0,
        "File::pushd" => 0,
        "Path::Tiny" => 0,
        "Test::DZil" => 0,
        "Test::Deep" => 0,
        "Test::Fatal" => 0,
        "Test::Kwalitee" => "1.21",
        "Test::More" => 0,
        "if" => 0
      }
    );


    my %fallback_build_requires = (
      "ExtUtils::MakeMaker" => 0,
      "File::Spec" => 0,
      "File::pushd" => 0,
      "Path::Tiny" => 0,
      "Test::DZil" => 0,
      "Test::Deep" => 0,
      "Test::Fatal" => 0,
      "Test::Kwalitee" => "1.21",
      "Test::More" => 0,
      "if" => 0
    );


    unless ( eval { Module::Build->VERSION(0.4004) } ) {
      delete $module_build_args{test_requires};
      $module_build_args{build_requires} = \%fallback_build_requires;
    }

    my $build = Module::Build->new(%module_build_args);

    $build->create_build_script;
}


Makefile.PL
# This Makefile.PL for Dist-Zilla-Plugin-Test-Kwalitee was generated by
# Dist::Zilla::Plugin::MakeMaker::Fallback 0.013
# and Dist::Zilla::Plugin::MakeMaker::Awesome 0.24.
# Don't edit it but the dist.ini and plugins used to construct it.

use strict;
use warnings;

BEGIN {
my %configure_requires = (
    'ExtUtils::MakeMaker' => '0',
    'Module::Build::Tiny' => '0.037',
);

my @missing = grep {
    ! eval "require $_; $_->VERSION($configure_requires{$_}); 1"
} keys %configure_requires;

if (not @missing)
{
    print "Congratulations, your toolchain understands 'configure_requires'!\n\n";
}
else
{
    $ENV{PERL_MM_FALLBACK_SILENCE_WARNING} or warn <<'EOW';
*** WARNING WARNING WARNING WARNING WARNING WARNING WARNING WARNING ***

If you're seeing this warning, your toolchain is really, really old* and you'll
almost certainly have problems installing CPAN modules from this century. But
never fear, dear user, for we have the technology to fix this!

If you're using CPAN.pm to install things, then you can upgrade it using:

    cpan CPAN

If you're using CPANPLUS to install things, then you can upgrade it using:

    cpanp CPANPLUS

If you're using cpanminus, you shouldn't be seeing this message in the first
place, so please file an issue on github.

If you're installing manually, please retrain your fingers to run Build.PL
when present instead.

This public service announcement was brought to you by the Perl Toolchain
Gang, the irc.perl.org #toolchain IRC channel, and the number 42.

----

* Alternatively, you are doing something overly clever, in which case you
should consider setting the 'prefer_installer' config option in CPAN.pm, or
'prefer_makefile' in CPANPLUS, to 'mb" and '0' respectively.

You can also silence this warning for future installations by setting the
PERL_MM_FALLBACK_SILENCE_WARNING environment variable.

EOW

    sleep 10 if -t STDIN && (-t STDOUT || !(-f STDOUT || -c STDOUT));
}
}

use 5.006;

use ExtUtils::MakeMaker;



my %WriteMakefileArgs = (
  "ABSTRACT" => "Release tests for kwalitee",
  "AUTHOR" => "Harley Pig <harleypig\@gmail.com>, Caleb Cushing <xenoterracide\@gmail.com>, Marcel Gruenauer <marcel\@cpan.org>, Kent Fredric <kentfredric\@gmail.com>, Karen Etheridge <ether\@cpan.org>",
  "CONFIGURE_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "Module::Build::Tiny" => "0.037"
  },
  "DISTNAME" => "Dist-Zilla-Plugin-Test-Kwalitee",
  "EXE_FILES" => [],
  "LICENSE" => "perl",
  "NAME" => "Dist::Zilla::Plugin::Test::Kwalitee",
  "PREREQ_PM" => {
    "Data::Section" => "0.004",
    "Dist::Zilla::File::InMemory" => 0,
    "Dist::Zilla::Plugin::MetaResources" => "4.300039",
    "Dist::Zilla::Role::FileGatherer" => 0,
    "Dist::Zilla::Role::PrereqSource" => 0,
    "Dist::Zilla::Role::TextTemplate" => 0,
    "Moose" => 0,
    "Sub::Exporter::ForMethods" => 0,
    "namespace::autoclean" => 0,
    "strict" => 0,
    "warnings" => 0
  },
  "TEST_REQUIRES" => {
    "ExtUtils::MakeMaker" => 0,
    "File::Spec" => 0,
    "File::pushd" => 0,
    "Path::Tiny" => 0,
    "Test::DZil" => 0,
    "Test::Deep" => 0,
    "Test::Fatal" => 0,
    "Test::Kwalitee" => "1.21",
    "Test::More" => 0,
    "if" => 0
  },
  "VERSION" => "2.11",
  "test" => {
    "TESTS" => "t/*.t"
  }
);


my %FallbackPrereqs = (
  "Data::Section" => "0.004",
  "Dist::Zilla::File::InMemory" => 0,
  "Dist::Zilla::Plugin::MetaResources" => "4.300039",
  "Dist::Zilla::Role::FileGatherer" => 0,
  "Dist::Zilla::Role::PrereqSource" => 0,
  "Dist::Zilla::Role::TextTemplate" => 0,
  "ExtUtils::MakeMaker" => 0,
  "File::Spec" => 0,
  "File::pushd" => 0,
  "Moose" => 0,
  "Path::Tiny" => 0,
  "Sub::Exporter::ForMethods" => 0,
  "Test::DZil" => 0,
  "Test::Deep" => 0,
  "Test::Fatal" => 0,
  "Test::Kwalitee" => "1.21",
  "Test::More" => 0,
  "if" => 0,
  "namespace::autoclean" => 0,
  "strict" => 0,
  "warnings" => 0
);


unless ( eval { ExtUtils::MakeMaker->VERSION(6.63_03) } ) {
  delete $WriteMakefileArgs{TEST_REQUIRES};
  delete $WriteMakefileArgs{BUILD_REQUIRES};
  $WriteMakefileArgs{PREREQ_PM} = \%FallbackPrereqs;
}

delete $WriteMakefileArgs{CONFIGURE_REQUIRES}
  unless eval { ExtUtils::MakeMaker->VERSION(6.52) };

WriteMakefile(%WriteMakefileArgs);



